### ISSUE TITLE
[hackerone]: blind cross chain signing should be prevented

### ISSUE THREAD

**Issue by kdenhartog** [brave/brave-browser#29798]
see https://hackerone.com/reports/1951016

**Comment by kjozwiak** [comment]
@diracdeltas should this be `release-notes/include` as it's a report from HackerOne?

**Comment by kjozwiak** [comment]
The above requires `1.51.105` or higher for `1.51.x` verification 👍

**Comment by diracdeltas** [comment]
@kjozwiak yeah, we can just use the PR title for the release notes: "Reject Solana SignTransaction/SignAllTransactions when blockhash is invalid (credit: https://hackerone.com/julianor)"

**Comment by srirambv** [comment]
Verification passed on Oppo Reno 5 with Android 13 running 1.51.109 x64 build
- Verified steps from [brave/brave-core#18200](https://github.com/brave/brave-core/pull/18200) using `https://pwgoom.csb.app`
- Verified clicking on `Sign Transaction (Invalid blockhash)` doesn't show the panel for approval but instead shows `[error] signTransactionWithInvalidBlockhash: {"code":-32603,"message":"Blockhash is invalid or can not be validated"}`
- Verified clicking on `Sign All Transaction (Invalid blockhash)` doesn't show the panel for approval but instead shows `[error] signAllTransactionsWithInvalidBlockhash: {"code":-32603,"message":"Blockhash is invalid or can not be validated"}`
- Verified `Sign Transaction (Invlalid blockhash)` and `Sign All Transactions (Invalid blockhash)` fails when both Solana Mainnet and Solana Devnet are set as active network
- Verified `Sign Transaction` and `Sign All Transactions (multiple)` works as expected

https://user-images.githubusercontent.com/17010094/235849415-95a910de-d406-4416-9cd2-b043d4350ba4.mp4
